src/main/java/org/codice/itest/TestExecutorTask.java

```diff
@@ -50,16 +50,22 @@ public void run() {
         try {
             diagnosticTest.setup();
             diagnosticTest.test();
-            diagnosticTest.cleanup();
             this.notify(testResultFactory.pass(testName, beforeTest, Instant.now()));
         } catch (AssertionError e) {
             this.notify(testResultFactory.fail(testName, e, beforeTest, Instant.now()));
         } catch (Throwable t) {
             this.notify(testResultFactory.error(testName, t, beforeTest, Instant.now()));
         }
+        finally {
+            try {
+                diagnosticTest.cleanup();
+            } catch (Exception e) {
+                this.notify(testResultFactory.error(testName, e, beforeTest, Instant.now()));
```

Comment:
I feel like we shouldn't use notify here. It would look odd to see the test pass and then error if cleanup isn't successful.

Reply:
Maybe we could store the TestResult into a variable and notify in a 2nd finally block after cleanup?